Objects/interpolationobject.c

```diff
@@ -0,0 +1,230 @@
+
+/*[clinic input]
+@classmethod
+Interpolation.__new__ as interpolation_new
+
+    value: object
+    expression: object(subclass_of='&PyUnicode_Type')
+    conversion: object(converter='_conversion_converter') = None
+    format_spec: object(subclass_of='&PyUnicode_Type', c_default='&_Py_STR(empty)') = ""
+[clinic start generated code]*/
+
+static PyObject *
+interpolation_new_impl(PyTypeObject *type, PyObject *value,
+                       PyObject *expression, PyObject *conversion,
+                       PyObject *format_spec)
+/*[clinic end generated code: output=6488e288765bc1a9 input=d91711024068528c]*/
+{
+    interpolationobject *self = PyObject_GC_New(interpolationobject, type);
+    if (!self) {
+        return NULL;
+    }
+
+    self->value = Py_NewRef(value);
+    self->expression = Py_NewRef(expression);
+    self->conversion = Py_NewRef(conversion);
+    self->format_spec = Py_NewRef(format_spec);
+    PyObject_GC_Track(self);
+    return (PyObject *) self;
+}
+
+static void
+interpolation_dealloc(PyObject *op)
+{
+    PyObject_GC_UnTrack(op);
+    Py_TYPE(op)->tp_clear(op);
+    Py_TYPE(op)->tp_free(op);
+}
+
+static int
+interpolation_clear(PyObject *op)
+{
+    interpolationobject *self = interpolationobject_CAST(op);
+    Py_CLEAR(self->value);
+    Py_CLEAR(self->expression);
+    Py_CLEAR(self->conversion);
+    Py_CLEAR(self->format_spec);
+    return 0;
+}
+
+static int
+interpolation_traverse(PyObject *op, visitproc visit, void *arg)
+{
+    interpolationobject *self = interpolationobject_CAST(op);
+    Py_VISIT(self->value);
+    Py_VISIT(self->expression);
+    Py_VISIT(self->conversion);
+    Py_VISIT(self->format_spec);
+    return 0;
+}
+
+static PyObject *
+interpolation_repr(PyObject *op)
+{
+    interpolationobject *self = interpolationobject_CAST(op);
+    return PyUnicode_FromFormat("%s(%R, %R, %R, %R)",
+                                _PyType_Name(Py_TYPE(self)), self->value, self->expression,
+                                self->conversion, self->format_spec);
+}
+
+static PyMemberDef interpolation_members[] = {
+    {"value", Py_T_OBJECT_EX, offsetof(interpolationobject, value), Py_READONLY, "Value"},
+    {"expression", Py_T_OBJECT_EX, offsetof(interpolationobject, expression), Py_READONLY, "Expression"},
+    {"conversion", Py_T_OBJECT_EX, offsetof(interpolationobject, conversion), Py_READONLY, "Conversion"},
+    {"format_spec", Py_T_OBJECT_EX, offsetof(interpolationobject, format_spec), Py_READONLY, "Format specifier"},
+    {NULL}
+};
+
+static PyObject*
+interpolation_reduce(PyObject *op, PyObject *Py_UNUSED(dummy))
+{
+    interpolationobject *self = interpolationobject_CAST(op);
+    return Py_BuildValue("(O(OOOO))", (PyObject *)Py_TYPE(op),
+                         self->value, self->expression,
+                         self->conversion, self->format_spec);
+}
+
+static PyMethodDef interpolation_methods[] = {
+    {"__reduce__", interpolation_reduce, METH_VARARGS,
```

Comment:
Question: shouldn't it be a `METH_NOARGS` instead?

Reply:
Yes, good catch!